tools/python/src/cnn_face_detector.cpp

```diff
@@ -59,12 +53,57 @@ class cnn_face_detection_model_v1
         // if the image was upscaled.
         for (auto&& d : dets) {
             d.rect = pyr.rect_down(d.rect, upsample_num_times);
-            rects.push_back(d.rect);
+            rects.push_back(d);
         }
 
         return rects;
     }
 
+    std::vector<std::vector<mmod_rect> > detect_mult (
+        boost::python::list& imgs,
+        const int upsample_num_times
+    )
+    {
+        pyramid_down<2> pyr;
+        std::vector<matrix<rgb_pixel> > dimgs;
+        dimgs.reserve(len(imgs));
+
+        for(auto i = 0; i < len(imgs); i++)
+        {
+            // Copy the data into dlib based objects
+            matrix<rgb_pixel> image;
+            object tmp = boost::python::extract<object>(imgs[i]);
+            if (is_gray_python_image(tmp))
+                assign_image(image, numpy_gray_image(tmp));
+            else if (is_rgb_python_image(tmp))
+                assign_image(image, numpy_rgb_image(tmp));
+            else
+                throw dlib::error("Unsupported image type, must be 8bit gray or RGB image.");
+
+            for(auto i = 0; i < upsample_num_times; i++)
+            {
+                pyramid_up(image);
+            }
+            dimgs.push_back(image);
+        }
+
+        auto dets = net(dimgs);
```

Comment:
This requires all the images to have the same size.  There is also an argument of the network that determines the mini-batch size.  For instance, if dimgs contained a million images, it wouldn't try to stuff 1 million images onto the GPU, it would be chopped up into batches of 128 (by default) and run.  The 128 value is the optional 2nd argument to net() and should also be an optional argument here, otherwise it will be impossible to run this function with large images since you will run out of GPU ram, but setting the batch size to something smaller would work fine. 

So this function needs two things.  A way to set the batch size, and a check that the user gave images that are all of the same size.  This check that all images are the same size is performed internally in the network as well, but the error message from there will probably confuse python users.